tools/approval-service/internal/approvalservice/sources/srv.go

```diff
@@ -0,0 +1,128 @@
+package sources
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net"
+	"net/http"
+
+	"golang.org/x/sync/errgroup"
+)
+
+// Server receives events from build systems.
+// It's typical for build systems to send events via webhooks.
+// This consolidates webhook events into a single server with a single endpoint.
+type Server struct {
+	addr     string
+	ln       net.Listener
+	handlers map[string]http.Handler
+
+	mux *http.ServeMux
+
+	log *slog.Logger
+}
+
+// ServerOpt is a functional option for the server.
+type ServerOpt func(*Server) error
+
+// WithAddress sets the address the server will listen on.
+func WithAddress(addr string) ServerOpt {
+	return func(s *Server) error {
+		if addr == "" {
+			return errors.New("address cannot be empty")
+		}
+		s.addr = addr
+		return nil
+	}
+}
+
+// WithListener sets the listener for the server.
+// This is an alternative to WithAddress for cases where it's more useful to set the listener directly (e.g. testing).
+func WithListener(ln net.Listener) ServerOpt {
```

Comment:
This does not seem to be used in the code or the tests. As this is `internal`, it is not a public library so unused code should be removed. If it is needed later, it can be added then.